app.py

```diff
@@ -19,23 +20,65 @@ def exportBag():
+    Aborts execution and returns HTTP body back to client if any 
+    request operation fails to succeed.
     """
     req_body = app.current_request.raw_body
     req_query_params = app.current_request.query_params
     req_headers = app.current_request.headers
     os.chdir('/tmp')
     with zipfile.ZipFile(io.BytesIO(req_body), 'r') as archive:
         archive.extractall()
-    payload = []
-    payload.append('/tmp/manifest/data/participant.tsv')
-    payload.append('/tmp/manifest/data/sample.tsv')
+    data = {
+        'participant': '/tmp/manifest/data/participant.tsv',
+        'sample': '/tmp/manifest/data/sample.tsv'
+    }
+    url = 'https://api.firecloud.org/api/workspaces'
     workspace = req_query_params['workspace']
     namespace = req_query_params['namespace']
-    url = 'https://api.firecloud.org/api/workspaces'
     auth = req_headers['Authorization']
-    manifest_io = ManifestIO(payload, url, workspace, namespace, auth)
-    statusCode = manifest_io.import_tsv_to_fc()
-    return Response(body=statusCode,
-                    status_code=200,
-                    headers={'Content-Type': 'application/json',
-                             'Accept': 'application/json'})
+    # Dict to collect status codes (or booleans) for successful operations.
+    statusCode = {
+        'status codes': {
+            'workspace exists': '',
+            'stood up workspace': '',
+            'participant upload': '',
+            'sample upload': ''
+        }
+    }
+    manifest_io = ManifestIO(data, url, workspace, namespace, auth, statusCode)
+    # Check whether the workspace exists in this namespace, otherwise create it.
+    r = manifest_io.workspace_exists()
+    if r.status_code != 200 and r.status_code != 404:
+        # Workspace neither exists, nor does the response indicate that it
+        # does not exist. Execution is aborted as some other reason is present,
+        # such as workspace name is invalid / not present / not authorized
+        # (401, 403) or backend is having issues (5xx) such as possibly
+        # an illegal workspace name (probably some 4xx)
+        return Response(body=r.text,
+                        headers={'Content-Type': 'text/plain'})
+    elif r.status_code == 404:
+        manifest_io.statusCode['status codes']['workspace exists'] = False
+        r_create_workspace = manifest_io.standup_workspace()
+        if r_create_workspace.status_code == 201:
+            manifest_io.statusCode['status codes']['stood up workspace'] = \
+                r_create_workspace.status_code
+    else:
+        # Workspace does exists.
+        manifest_io.statusCode['status codes']['workspace exists'] = \
+            r.status_code
+        manifest_io.statusCode['status codes']['stood up workspace'] = False
+
+    # Order keys of dictionary data alphabetically to ensure that participant
+    # is uploaded before sample is uploaded.
+    data_ordered = OrderedDict(sorted(data.items(), key=lambda t: t[0]))
```

Comment:
I forgot to comment on this earlier, but I think it's happenstance that participant is alphabetically before sample, and I don't think your order should depend on it.

Aren't there other entity types as well (pairs, sample sets), that might get added at some point? If so, do they happen to follow this alphabetical order?

I think you this code just needs to know the order, probably hard-coded, e.g., get the value for the particpant, then get the value for the sample.